sspl_test/alerts/self_hw/test_node_bmc_conn.py

```diff
@@ -0,0 +1,98 @@
+# If there is no failure, that would mean connectivity is OK
+
+import os
+import re
+import time
+import socket
+
+from framework.utils.conf_utils import (
+    Conf, GLOBAL_CONF, ENCLOSURE, MACHINE_ID, BMC_IP_KEY, BMC_USER_KEY, BMC_SECRET_KEY)
+from cortx.utils.process import SimpleProcess
+from cortx.utils.security.cipher import Cipher
+from cortx.utils.validator.v_bmc import BmcV
+
+
+CHANNEL_PROTOCOL = "Channel Protocol Type"
+FIRMWARE_VERSION = "Firmware Revision"
+
+def init(args):
+    pass
+
+def test_bmc_config(args):
+    """Check if BMC configuration are valid."""
+    pkey_file = "/root/.ssh/id_rsa_prvsnr"
+    if not os.path.exists(pkey_file):
+        print("\tSkipping test_bmc_config.. pkey file not found.")
+        return
+    bmc_ip = Conf.get(GLOBAL_CONF, BMC_IP_KEY)
+    bmc_user = Conf.get(GLOBAL_CONF, BMC_USER_KEY)
+    bmc_secret = Conf.get(GLOBAL_CONF, BMC_SECRET_KEY)
+    bmc_key = Cipher.generate_key(MACHINE_ID, "server_node")
+    bmc_passwd = Cipher.decrypt(
+        bmc_key, bmc_secret.encode("utf-8")).decode("utf-8")
+    BmcV().validate(
+        "accessible", [socket.getfqdn(), bmc_ip, bmc_user, bmc_passwd])
+
+def test_bmc_firmware_version(args):
+    """Check if BMC firmware version is 1.71."""
+    cmd = "ipmitool bmc info"  # Firmware Revision : 1.71
+    expected_ver = "1.71"
+    version_found = None
+    res_op, res_err, res_rc = SimpleProcess(cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        search_res = re.search(
+            r"%s[\s]+:[\s]+([\w.]+)(.*)" % FIRMWARE_VERSION, res_op)
+        if search_res:
+            version_found = search_res.groups()[0]
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+    if expected_ver != version_found:
+        print("UNEXPECTED BMC FIRMWARE VERSION FOUND.")
+        print("Expected: %s" % expected_ver)
+        print("Found: %s" % version_found)
+
+def test_bmc_accessible_through_kcs(args):
+    """Check if BMC channel type is KCS."""
+    cmd = "sudo ipmitool channel info"  # Channel Protocol Type : KCS
+    expected_channel = "KCS"
+    channel_found = None
+    res_op, res_err, res_rc = SimpleProcess(cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        search_res = re.search(
+            r"%s[\s]+:[\s]+(\w+)(.*)" % CHANNEL_PROTOCOL, res_op)
+        if search_res:
+            channel_found = search_res.groups()[0]
+        if expected_channel != channel_found:
+            print("UNEXPECTED BMC CHANNEL TYPE FOUND.")
+            print("Expected: %s" % expected_channel)
+            print("Found: %s" % channel_found)
+    else:
+        res_err = res_err.decode()
+        kcs_errors = ("could not find inband device", "driver timeout")
+        if not any(err for err in kcs_errors if err in res_err):
+            raise Exception("ERROR: %s" % res_err.decode())
+
+test_list = [
+    test_bmc_config,
+    test_bmc_firmware_version,
+    test_bmc_accessible_through_kcs
```

Comment:
>ipmitool chassis selftest
Self Test Results    : passed

Can we add this test too in the list and probably rename this test file as bmc tests ?

Reply:
Added.